src/EncryptionBehavior.php

```diff
@@ -330,22 +332,24 @@ protected static function insertEncryptedColumnsDeclaration(string &$script, int
      * @param string $script
      * @param string $columnPhpName
      * @param bool $isSearchable
-     *
-     * @throws \Exception
+     * @param bool $hasColumnBlobType
      *
      * @return void
      */
-    protected function addEncryptionToSetter(string &$script, string $columnPhpName, bool $isSearchable): void
-    {
-        $setterLocation = strpos($script, "set$columnPhpName");
-
-        if ($setterLocation === false) {
-            throw new Exception(sprintf(
-                'The `%s()` method was not found in the script.',
-                "set$columnPhpName",
-            ));
+    protected function addEncryptionToSetter(
+        string &$script,
+        string $columnPhpName,
+        bool $isSearchable,
+        bool $hasColumnBlobType
+    ): void {
+        $setterLocation = $this->getMethodLocation($script, "set$columnPhpName");
+
+        if ($hasColumnBlobType) {
+            $paramAnnotationLocation = strpos($script, 'param resource', $setterLocation - 300);
```

Comment:
Perhaps we should implement a method that finds the nearest previous '}' before the name of the method, in order to accurately determine the phpdoc in which wehave to make a replacement.